libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java

```diff
@@ -95,27 +103,44 @@ private void muteTts() {
   }
 
   void setInstructionListener(final InstructionListener instructionListener) {
-    textToSpeech.setOnUtteranceProgressListener(new UtteranceProgressListener() {
-      @Override
-      public void onStart(String utteranceId) {
-        if (instructionListener != null) {
-          instructionListener.onStart();
-        }
-      }
+    this.instructionListener = instructionListener;
 
-      @Override
-      public void onDone(String utteranceId) {
-        if (instructionListener != null) {
-          instructionListener.onDone();
-        }
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+      textToSpeech.setOnUtteranceCompletedListener(new Api14UtteranceListener());
+    } else {
+      textToSpeech.setOnUtteranceProgressListener(new UtteranceListener());
+    }
+  }
+
+
+  class Api14UtteranceListener implements TextToSpeech.OnUtteranceCompletedListener {
```

Comment:
The listeners extracted look 💯 - would we be able to extract them into their own class files to further clean up `AndroidSpeechPlayer`?